controllers/mustgather/mustgather_controller.go

```diff
@@ -367,23 +374,78 @@ func (r *MustGatherReconciler) addFinalizer(reqLogger logr.Logger, m *mustgather
 	return nil
 }
 
-func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.MustGather) (*unstructured.Unstructured, error) {
+func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.MustGather) (*batchv1.Job, error) {
 	version, err := r.getClusterVersionForJobTemplate("version")
 	if err != nil {
 		return nil, fmt.Errorf("failed to get cluster version for job template: %w", err)
 	}
 
-	jobTemplate, err = initializeTemplate(version)
+	jobTemplate, err := initializeTemplate(version)
 	if err != nil {
 		log.Error(err, "unable to initialize job template")
-		return &unstructured.Unstructured{}, err
+		return nil, err
 	}
-	unstructuredJob, err := templates.ProcessTemplate(context.TODO(), instance, jobTemplate)
+
+	job, err := processJobTemplate(jobTemplate, instance)
 	if err != nil {
-		log.Error(err, "unable to process", "template", jobTemplate, "with parameter", instance)
-		return &unstructured.Unstructured{}, err
+		log.Error(err, "unable to process job template")
+		return nil, err
 	}
-	return unstructuredJob, nil
+
+	return job, nil
+}
+
+func processJobTemplate(jobTemplate string, instance *mustgatherv1alpha1.MustGather) (*batchv1.Job, error) {
```

Comment:
Do we need the jobTemplate YAML for anything? Should we just default values in the struct itself?

```go
job := &batchv1.Job{
  ObjectMeta: metav1.ObjectMeta{
    Name:      instance.ObjectMeta.Name,
    Namespace: instance.ObjectMeta.Namespace,
  },
  Spec: {
    // The rest of the default/non-default values
  }

```

Reply:
I'm fully on board with this approach, but it's more work. Long term I think it's better and am more than happy to go through with this.